### DISCUSSION TITLE
Add PluginFactory functionality to Craft CLI

### DISCUSSION THREAD

**Discussion by dougstjohn** [craftcms/cms#8786] [Ideas]
https://pluginfactory.io/ is currently the easiest way to generate a plugin or module for Craft and much easier than creating one from scratch.

### This issue is to add CLI commands to:
- generate a plugin or module (and define which components you want generated)
- add a fieldtype/controller/settings/etc to an existing plugin or module
- auto-install the plugin or module after being generated (via composer)

### Generating a plugin or module:
`./craft generate/plugin dougstjohn/my-plugin`
`./craft generate/module dougstjohn/my-module`
`./craft generate/module dougstjohn/my-module --prompt`
`./craft generate/module dougstjohn/my-module --components=controller,settings,model`

### Generating a component into a plugin or module:
`./craft generate/fieldtype dougstjohn/my-plugin <name>`
`./craft generate/model dougstjohn/my-module <name>`
`./craft generate/record dougstjohn/my-module <name>`
...etc

**Comment by dougstjohn** [comment]
FYI @nystudio107

**Comment by piotrpog** [comment]
Yes please!
